File: MarketAPI/MarketAPI/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MarketAPI.Models;

[Table("Products")]
public class Product
{
    [Key]
    public int Id { get; set; }

    [Required, StringLength(100)]
    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    public int? MeasurementUnitId { get; set; }

    public string? ImageUrl { get; set; }

    [Required]
    public bool IsActive { get; set; } = true;

    [Required]
    public int CategoryId { get; set; }

    // Navegación
    public Category Category { get; set; } = null!;
    public MeasurementUnit? MeasurementUnit { get; set; }
}

File: MarketClient/src/types/product.ts
export interface Product {
  id: number;
  name: string;
  description?: string;
  measurementUnitId: number;
  imageUrl?: string;
  isActive: boolean;
  categoryId: number;
  categoryName?: string;
  measurementUnitName?: string;
}

export interface CreateProductRequest {
  name: string;
  description?: string;
  measurementUnitId: number;
  imageUrl?: string;
  categoryId: number;
}

export interface UpdateProductRequest {
  name: string;
  description?: string;
  measurementUnitId: number;
  imageUrl?: string;
  isActive: boolean;
  categoryId: number;
}
